src/distribution/cauchy.rs

```diff
@@ -0,0 +1,238 @@
+use distribution;
+use source::Source;
+
+/// A Cauchy distribution.
+///
+/// A Cauchy distribution (aka Lorentz or Cauchy–Lorentz distribution) is a continuous
+/// probability distribution with a location parameter `loc`, a scale parameter `gamma > 0`,
+/// and the following probability density function:
+///
+/// `p(x) = const / (1 + ((x - loc) / gamma)^2)`.
+///
+/// A Cauchy distribution is long tailed and has no well-defined mean or variance. It is
+/// unimodal with its mode at `loc`, around which it is symmetric. The ratio of two
+/// independent Gaussian distributed random variables is Cauchy distributed.
+///
+/// See [Wikipedia article on Cauchy
+/// distribution](https://en.wikipedia.org/wiki/Cauchy_distribution).
+#[derive(Clone, Copy, Debug)]
+pub struct Cauchy {
+    loc: f64,
+    gamma: f64,
+}
+
+impl Cauchy {
+    /// Create a Cauchy distribution with location `loc` and scale `gamma`.
+    ///
+    /// It should hold that `gamma > 0`.
+    #[inline]
+    pub fn new(loc: f64, gamma: f64) -> Self {
+        should!(gamma > 0.0);
+        Cauchy { loc, gamma }
+    }
+
+    // Return the location parameter
+    #[inline(always)]
+    pub fn loc(&self) -> f64 {
+        self.loc
+    }
+
+    // Return the scale parameter
+    #[inline(always)]
+    pub fn gamma(&self) -> f64 {
+        self.gamma
+    }
+}
+
+impl distribution::Continuous for Cauchy {
+    #[inline]
+    fn density(&self, x: f64) -> f64 {
+        // Divisions are expensive, so we implement the PDF with only a single division.
+        use std::f64::consts::PI;
+        let deviation = x - self.loc;
+        self.gamma / (PI * (self.gamma * self.gamma + deviation * deviation))
+    }
+}
+
+impl distribution::Distribution for Cauchy {
+    type Value = f64;
+
+    #[inline]
+    fn distribution(&self, x: f64) -> f64 {
+        use std::f64::consts::FRAC_1_PI;
+        FRAC_1_PI * ((x - self.loc) / self.gamma).atan() + 0.5
+    }
+}
+
+impl distribution::Entropy for Cauchy {
+    #[inline]
+    fn entropy(&self) -> f64 {
+        (std::f64::consts::PI * 4.0 * self.gamma).ln()
+    }
+}
+
+impl distribution::Inverse for Cauchy {
+    /// Due to finite precision of arithmetic operations, the current implementation of
```

Comment:
Not sure about this, as other distributions handle this:

https://github.com/stainless-steel/probability/blob/master/src/distribution/gaussian.rs#L232

It might make sense to treat edge cases separately.

Reply:
agreed.